From 2d7bf366acf6602531067ca53e9948d1012ced0b Mon Sep 17 00:00:00 2001
From: Bharat Kumar Gogada <bharat.kumar.gogada@xilinx.com>
Date: Mon, 13 Jan 2020 15:26:25 +0530
Subject: [PATCH 0829/1852] PCI: XDMA PL PCIe: Move to struct pci_host_bridge
 IRQ mapping functions

commit 0d0b658034446943818787b066297249c9a77411 from
https://github.com/Xilinx/linux-xlnx.git

Add map/swizzle IRQs hooks to move the IRQ allocation into
core code and stop relying on arch specific callbacks.

Signed-off-by: Bharat Kumar Gogada <bharat.kumar.gogada@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: not-upstreamable
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/pci/controller/pcie-xdma-pl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/pci/controller/pcie-xdma-pl.c b/drivers/pci/controller/pcie-xdma-pl.c
index a2d20eb15c5b..6e6837a99257 100644
--- a/drivers/pci/controller/pcie-xdma-pl.c
+++ b/drivers/pci/controller/pcie-xdma-pl.c
@@ -641,6 +641,8 @@ static int xilinx_pcie_probe(struct platform_device *pdev)
 	bridge->sysdata = port;
 	bridge->busnr = port->root_busno;
 	bridge->ops = &xilinx_pcie_ops;
+	bridge->map_irq = of_irq_parse_and_map_pci;
+	bridge->swizzle_irq = pci_common_swizzle;
 
 	err = pci_scan_root_bus_bridge(bridge);
 	if (err)
-- 
2.31.1

